### ISSUE TITLE
stmt.h contains components stdin and stdout - they cause confusion

### ISSUE THREAD

**Issue by arjenmarkus** [CodethinkLabs/ofc#27]
When I tried to compile the code using GCC 5.3.0 on Cygwin, I got error messages about the use of stdin and stdout as component names in stmt.h

I solved that by appending an X to those names (also in read.c and write.c). Then it worked.

**Comment by ben-brewer-codethink** [comment]
This sounds like an issue with the standard library you're using, can you tell me if adding the following works?

```
#ifdef stdin
#undef stdin
#endif
#ifdef stdout
#undef stdout
#endif
```

**Comment by arjenmarkus** [comment]
Hi Ben,

I inserted these lines in stmt.h just above the "typedef enum" at line 19.
The build succeeds without any complaints (or they have scrolled off the
screen :)). So this does seem a solution.

Regards,

Arjen

2016-09-05 14:40 GMT+02:00 Ben Brewer notifications@github.com:

**Comment by ben-brewer-codethink** [comment]
Cool, fixed in: 8de524fdca0775cde97db4d504add2a96f3bca39

Thanks for reporting the bug.

**Comment by arjenmarkus** [comment]
You're welcome

BTW: I have ignored the error messages about my directory not being a git
repository (I took the zip file instead of clone). they looked menacing,
but as the build succeeded, I decided it was not worth the trouble.

Regards,

Arjen

2016-09-05 15:26 GMT+02:00 Ben Brewer notifications@github.com:

**Comment by arjenmarkus** [comment]
Hi Ben,

are you interested in more error reports? I just tried OFC on an
implementation of a TOMS algorithm and I found a bunch of odd things.

I can either put them in error reports in the project or send them to you
by e-mail.

Regards,

Arjen

2016-09-05 15:34 GMT+02:00 Arjen Markus arjen.markus895@gmail.com:

**Comment by ben-brewer-codethink** [comment]
We're always interested in error reports! They really help us to guarantee some level of quality.

It's an open project so it's good for people to see what errors there are and when they're resolved.

My preference is that errors are reported on github, however if there's code or discussions you don't want to publish in the open then I'm happy to handle those in a private e-mail.

**Comment by arjenmarkus** [comment]
No problem - the software I have been looking at is open source (
www.netlib.org), I just used OFC on it to see what it would come up with. I
have some serious doubts about the actual implementation. I just noticed in
the report from OFC a few things that do not correspond to a correct
interpretation of the FORTRAN 77 source code.

I will create the various error reports - it should prove interesting to
solve these.

Regards,

Arjen

2016-09-06 14:46 GMT+02:00 Ben Brewer notifications@github.com:
